packages/snap-preact-demo/tests/cypress/e2e/tracking/track.cy.js

```diff
@@ -1,382 +1,237 @@
+			expect(context).to.have.property('initiator').to.be.a('string');
+			expect(context).to.have.property('pageLoadId').to.be.a('string');
+			expect(context).to.have.property('pageUrl').to.be.a('string');
+			expect(context).to.have.property('sessionId').to.be.a('string');
+			expect(context).to.have.property('timestamp').to.be.a('string');
+			expect(context).to.have.property('userAgent').to.be.a('string');
+			expect(context).to.have.property('userId').to.be.a('string');
+
+			expect(response.body).to.have.property('success').to.equal(true);
+		});
 	});
-	it('tracked shopper login', () => {
+
+	it('tracked search render, impression, clickthrough', () => {
 		cy.visit('https://localhost:2222');
 
-		cy.waitForBundle().then((searchspring) => {
-			expect(searchspring).to.exist;
+		cy.wait(`@beacon2/search/render`).then(({ request, response }) => {
+			expect(response.body).to.have.property('success').to.equal(true);
+
+			const { context, data } = JSON.parse(request.body);
+			expect(context).to.be.an('object');
+			expect(data).to.have.property('results').to.be.an('array').length.greaterThan(0);
+			expect(data.results[0]).to.have.property('uid').to.be.a('string');
+			expect(data).to.have.property('merchandising').to.be.an('object');
+			expect(data).to.have.property('pagination').to.be.an('object');
+			expect(data).to.have.property('sort').to.be.an('array').length.greaterThan(0);
+			expect(data).to.have.property('q').to.be.a('string');
 		});
 
-		// wait for first login event
-		cy.wait(`@${BeaconType.LOGIN}`).then(() => {
-			// initial init will send a login event for the shopper due to integration script variables
-
-			cy.wait(1000);
-			const shopperId = 'snaptest';
-			cy.get('#login').click();
-			cy.get('#login-modal').find('input').type(shopperId);
-			cy.get('#login-modal').find('button').click();
-
-			// test of new login using modal and tracker function
-			cy.wait(`@${BeaconType.LOGIN}`).should((interception) => {
-				expect(interception.state).to.equal('Complete');
-				// expect(interception.response.body).to.have.property('success').to.equal(true);
-
-				const beacon = interception.request.body.filter((event) => event.type === BeaconType.LOGIN)[0];
-				expect(beacon.category).to.equal(BeaconCategory.PERSONALIZATION);
-				expect(beacon.type).to.equal(BeaconType.LOGIN);
-				expect(beacon.event).to.be.an('object').include.key('shopperId').include.key('userId');
-				expect(beacon.context).to.be.an('object').include.key('shopperId');
-				expect(beacon.context.shopperId).to.equal(shopperId);
-			});
+		cy.scrollTo(0, 500); // Scroll down to trigger impressions
+
+		cy.wait(`@beacon2/search/impression`, { timeout: 10000 }).then(({ request, response }) => {
+			expect(response.body).to.have.property('success').to.equal(true);
+
+			const { data } = JSON.parse(request.body);
+			expect(data).to.have.property('results').to.be.an('array').length.greaterThan(0);
```

Comment:
We should be explicit instead of just greater than zero - exactly how many results do we expect to be here?